### DISCUSSION TITLE
Welcome to RBFOpt Discussions!

### DISCUSSION THREAD

**Discussion by gnannicini** [coin-or/rbfopt#34] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. 

You can help the community simply by asking and/or answering questions!

**Comment by ThiThoiTran** [comment]
Hello, I would like to ask about providing user initial designs to RBFopt. The manual has shown there is a bool variable do_init_strategy (switch True or False: True when rbfopt creates its own design, False when rbfopt does not create initial design but taking the user points), but I could not see it in the files. I tried to declare it through the interface file rbfopt_cl_interface.py: 
      intset.add_argument('--do_init_strategy', action = 'store', dest = 'do_init_strategy',
                        default = True,type = ast.literal_eval,
                        help = 'DoE generated by algorithm or ' +
                        'provided by the user')
and add the command -f interpolation_file (.txt format) --do_init_strategy False
It added the providing initial design, but at the beginning, the algorithm still creates its own design which contains n+1 points.   Could you please help me? Thank you

**Reply by gnannicini** [reply]
Hi,
if you are using the command line interface, there is currently no way to set do_init_strategy to False. This is because the algorithm can fail if the initial design does not satisfy certain requirements. So if you use the command line interface, any provided initial design will simply be added to the one generated by RBFOpt: this is safer, and should never fail regardless of the points you provide.

You can set do_init_strategy to False if you use the API to call RBFOpt, and you're sure that this is what you want to do. The API also allows you to set initial points, of course. So if this is your goal, I suggest you write a short piece of code to create your RbfoptBlackBox, RbfoptSettings, and an RbfoptAlgorithm setting do_init_strategy=False.

Hope this helps.

**Reply by ThiThoiTran** [reply]
Thank you. Actually, I want to test the effect of different DoEs (user define the number of points and set of points) in optimization, I am working on trust-region method, DFOb-TR (extended to mixed variables from the work of Andrew  R.Conn).  Since the analysis has shown a big impact of DoEs in the results of running the optimization, so I and my supervisors decided to develop new DoE methods in mixed space (using embedding kernel approach).  We tested these new DoEs with EGO, and DFOb-TR, and we really want to test them with RBFopt also!!  Thank you again, I will try as your suggestion.

**Reply by ThiThoiTran** [reply]
Great, it works, I have succeeded to run rbfopt with the user providing initial design in pycharm:) Thank you again!!
